### PR TITLE
MM-45754 - Calls: Add current call bar to thread screen

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Add the current call bar to the thread screen
- The floating container doesn't need the header_height added for the thread screen -- is this expected @enahum ?

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-45754

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
- Android: 12, Pixel 6
- iOS: 15.3.1, iPhone 7 plus

#### Screenshots
image

#### Release Note
```release-note
Calls: Current call bar is now shown on the Thread screen
```

### CODE REVIEW

File: app/screens/channel/index.tsx [hunk 1]
```diff
@@ -20,10 +20,6 @@ type EnhanceProps = WithDatabaseArgs & {
 
 const enhanced = withObservables([], ({database, serverUrl}: EnhanceProps) => {
     const channelId = observeCurrentChannelId(database);
-    const isCallsPluginEnabled = observeCallsConfig(serverUrl).pipe(
```
Comment (cpoile) [inline app/screens/channel/index.tsx:23]: Removing this; it was a mistake to add it -- it doesn't matter if the calls plugin is enabled for the current server, the user could still be on a call from a different server.

**Review by enahum — COMMENTED** [review]
> The floating container doesn't need the header_height added for the thread screen -- is this expected @enahum ?

Yes, is expected. On the channel screen we use a custom component as the navigation bar while on the thread we use the native one, that is the difference.

One other thing to note is, when we tap on the calls banner in a thread, it will open the call screen, iirc there is an option to go to the thread, how is this being handled to avoid mounting the same screens multiple times?

**Comment by cpoile** [comment]
> when we tap on the calls banner in a thread, it will open the call screen, iirc there is an option to go to the thread, how is this being handled to avoid mounting the same screens multiple times?

@enahum You're right, there's a `Call Thread` action in the `More` menu of the call screen. It does bring up a new thread screen. We could track if we're coming from that thread, and pop the call screen instead of opening another thread screen. That would solve it.

But, thinking about UX: would that be confusing for the user? They would swipe back, thinking they would get back to the call screen, but instead they go back to whatever screen they were on before bringing up the thread the first time. (3 screens ago, in their mind). 

Basically: If I was in the call screen, and went to the call thread, and then tapped the back arrow (or swiped left), I would expect to go back to the call screen.

**Comment by enahum** [comment]
> Okay, it looks like iOS and Android behave differently here. In iOS you can go however many stacks deep you want: Thread -> Call Screen -> Thread -> Call Screen, so on. In Android you can only have one Thread on the screen stack. So if you entered the Call Screen from a Thread, I guess we're going to have to pop the Call Screen and the previous Thread screen before navigating to the call's Thread screen? What do you think, is there a solution I'm missing here?

You have two options that will work on both platforms
1. Update the Thread screen props and pop the calls screen
2. Push another Thread screen but set it with a different ID, the problem with this approach is that you need to look around the code base where we try and determine if there is a Thread screen already mounted and what are we doing in those cases.

File: app/screens/navigation.ts [hunk 2]
```diff
@@ -421,6 +421,7 @@ export async function dismissAllModalsAndPopToScreen(screenId: string, title: st
         }
         try {
             await Navigation.popTo(screenId, mergeOptions);
+            await Navigation.updateProps(screenId, passProps);
```
Comment (cpoile) [inline app/screens/navigation.ts:424]: added

Comment (enahum) [reply app/screens/navigation.ts:424]: This looks ok, just wondering if this should be optional so that if there is no need to update the props and this function is called we could then avoid unnecessary re-renders

File: app/screens/thread/index.tsx [hunk 3]
```diff
@@ -3,17 +3,27 @@
...
+    );
+
     return {
         rootPost: observePost(database, rootId),
+        isInACall,
     };
 });
 
-export default withDatabase(enhanced(Thread));
+export default withDatabase(withServerUrl(enhanced(Thread)));
```
Comment (enahum) [inline app/screens/thread/index.tsx:29]: Where is the serverUrl being used?

File: app/screens/navigation.ts (hunk 2 above)
Comment (cpoile) [reply app/screens/navigation.ts]: Ah, good point. Fixed -- will call update props if we're given passProps.

File: app/screens/thread/index.tsx (hunk 3 above)
Comment (cpoile) [reply app/screens/thread/index.tsx]: Sorry, must have been refactored out and I forgot to remove it. :) Fixed.

**Review by enahum — APPROVED** [review]

**Comment by cpoile** [comment]
Gentle ping @streamer45 when you're able.

**Review by streamer45 — APPROVED** [review]
Apologies for the late approval. Looks good :+1:
